src/main/java/reposense/util/FileUtil.java

```diff
@@ -55,55 +64,78 @@ public static void deleteDirectory(String root) throws IOException {
         }
     }
 
-    public static void unzip(ZipInputStream zipInput, String destinationFolder) {
-        Path directory = Paths.get(destinationFolder);
-
-        // buffer for read and write data to file
-        byte[] buffer = new byte[2048];
+    /**
+     * Zips all the files of {@code fileTypes} contained in {@code sourceAndOutputPath} directory into the same folder.
+     */
+    public static void zip(Path sourceAndOutputPath, String... fileTypes) {
+        FileUtil.zip(sourceAndOutputPath, sourceAndOutputPath, fileTypes);
+    }
 
-        try {
-            Files.createDirectories(directory);
-            ZipEntry entry = zipInput.getNextEntry();
+    /**
+     * Zips all the {@code fileTypes} files contained in the {@code sourcePath} and its subdirectories.
+     * Creates the zipped {@code ZIP_FILE} file in the {@code outputPath}.
+     */
+    public static void zip(Path sourcePath, Path outputPath, String... fileTypes) {
+        try (
+                FileOutputStream fos = new FileOutputStream(outputPath + File.separator + ZIP_FILE);
+                ZipOutputStream zos = new ZipOutputStream(fos)
+        ) {
+            Set<Path> allFiles = getFilePaths(sourcePath, fileTypes);
+            for (Path path : allFiles) {
+                String filePath = sourcePath.relativize(path.toAbsolutePath()).toString();
+                String zipEntry = Files.isDirectory(path) ? filePath + File.separator : filePath;
+                try (InputStream is = Files.newInputStream(path)) {
+                    zos.putNextEntry(new ZipEntry(zipEntry.replace("\\", "/")));
+                    int length;
+                    while (Files.isRegularFile(path) && (length = is.read(buffer.array())) > 0) {
```

Comment:
This implementation is smelly.

Why open InputStream for both `folder` and `file`?

You can restructure the block by moving `Files.isRegularFile(path)` upwards.

If the reason for `InputStream` is to write to file, then the question is - Do I need to open `InputStream` for folder?